### ISSUE TITLE
[Bug] NX Springboot app generation in an empty workspace requires Angular dependencies

### ISSUE THREAD

**Issue by Kwandes** [tinesoft/nxrocks#42]
Plugin Name and Version:
@nxrocks/nx-spring-boot@1.3.1

Nx Version (`nx --version`):
11.4.0

Node Version (`node -v`):
Current LTS: 14.16.0

NPM  Version (`npm -v`):
6.14.11

Angular CLI version (`ng --version`):
None, it's an empty NX project

#### Expected Behaviour
When I run `nx g @nxrocks/nx-spring-boot:app java-api` and fill out the options, a new java springboot app gets generated

#### Actual Behaviour
When I run `nx g @nxrocks/nx-spring-boot:app java-api` and fill out the options, the generation fails with an error:
```sh
Cannot find module '@angular-devkit/core'
Require stack:
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\src\commands\ngcli-adapter.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\src\commands\generate.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\index.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\cli\lib\init-local.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\cli\bin\nx.js
- C:\Users\kwandes\AppData\Roaming\nvm\v14.16.0\node_modules\nx\node_modules\@nrwl\cli\bin\nx.js
- C:\Users\kwandes\AppData\Roaming\nvm\v14.16.0\node_modules\nx\bin\nx.js
```

After that, I install the missing dependency with `npm install @angular-devkit/core`.
The generation fails again with 
```sh
Cannot find module '@angular-devkit/schematics/tools'
Require stack:
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\src\commands\ngcli-adapter.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\src\commands\generate.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\tao\index.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\cli\lib\init-local.js
- O:\Programming\testing\troubleshoot-nvm\node_modules\@nrwl\cli\bin\nx.js
- C:\Users\kwandes\AppData\Roaming\nvm\v14.16.0\node_modules\nx\node_modules\@nrwl\cli\bin\nx.js
- C:\Users\kwandes\AppData\Roaming\nvm\v14.16.0\node_modules\nx\bin\nx.js
```
Module '@angular-devkit/schematics/tools' can not be installed via npm

#### Steps to reproduce the behaviour

Run `nvm install 14.16.0`
Run `nvm use 14.16.0`

Run `npx create-nx-workspace@latest`
 - name: troubleshoot
 - type: empty
 - nx cloud: yes
 
Run `npm install -g nx@11.4.0`
Run `npm install --save-dev @nxrocks/nx-spring-boot`

Run `nx g @nxrocks/nx-spring-boot:app java-api`
Fill out the info, the generation will fail

Run ´npm install @angular-devkit/core´
Run `nx g @nxrocks/nx-spring-boot:app java-api`
Fill out the info, the generation will fail

### Notes

- I suspect this is caused by the workspace being an empty project instead of an angular one
- The issue does not occur in an NX workspace with preset Angular
- The Username in  windows path ` C:\Users\` has been replaced with `kwandes`

**Comment by tinesoft** [comment]
Hi again @Kwandes!

Thanks for the continuing use and helpful bug reporting on the plugin, much appreciated 👍🏾 

The error you mentioned is due to [this recent change](https://github.com/nrwl/nx/pull/4610) in Nx workspace starting v11.2.6. In fact the dependencies `@angular-devkit/{core,architect,schematics}`, that this nx plugin (like most) is using internally, have been removed from Nx, as they are migrating away from `@angular-devkit/*` and its schematics (resp. builders) API to their own implementation `@nrwl/devkit` and its generators (resp. executors) API.

I'm currently migrating the plugin to be compatible with this new API (in the upcoming v2.x.x).
That should fixed the problem. I will add a Nx compatibility matrix on the README page once done.

Until then, I'm afraid you must stick to Nx workspace < v11.2.6.

**Comment by tinesoft** [comment]
Fixed in [v2.0.0](https://github.com/tinesoft/nxrocks/releases/tag/nx-spring-boot%2Fv2.0.0)
